orderer/common/cluster/block_fetcher_test.go

```diff
@@ -349,6 +361,242 @@ func TestBlockFetcherBFTBehaviorSuspicionNoBlockWithhold(t *testing.T) {
 	require.Equal(t, (*common.Block)(nil), block)
 }
 
+func TestBlockFetcherBFTBehaviorSuspicionListFull(t *testing.T) {
+	// suspicion list of block fetcher (bf.suspects) can hold max MaxByzantineNodes entries, when it is full,
+	// last entry is evicted to accomodate the new entry.
+	// 2 malicious orderers
+	// bf.MaxByzantineNodes = 1
+	// Try to pull blocks form orderer, it witholds blocks,
+	// add it to suspicion list which is now full and check next, it also withold blocks
+	// add it to list, which is full, so evict one element from the list
+	// The first endpoint we try to pull Blocks witholds blocks,
+	// the PullBlock should then suspect the endpoint and probe other endpoints to confirm the suspicion.
+	// then this endpoint should be added to suspects list and blockfetcher should shuffle the endpoint and
```

Comment:
how is the suspect list logic described here being tested?

Reply:
MaxByzantineNodes=1, we can pull from nodes (node1, node2, node1, node3) in this order. First node1 is byzantine, so we add it to suspect list of size 1, then we try to pull from node2, it is also byzantine, so we update it to suspect list, evicting node1 from suspect list  then we try to pull from node 1 again, if node1 was successfully evicted from suspect list, we should be able to pull from node1(not byzantine now), if it is still in suspect list(due to bug in code), we should not be able to pull from node1 and should shuffle to pull from node3. So we pull block and check if it is pulled from node1